### PR TITLE
Updated contribution guidelines and developer steps

### PR DESCRIPTION
**Pull request by cullywakelin**
This documentation repository is open source and contributions are welcome. The README provides instructions on how to contribute and what to expect when your changes are merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,122 @@
...
+Check out their [installation requirements](https://v2.docusaurus.io/docs/installation#requirements) for tooling prerequisites.
+
+### Step 1: Initialize yarn
+
+The first step is to initialize the use of Yarn to build the site locally.
+
+From your terminal, change directory to the project root and run:
+
+```
+$ yarn
```
Comment (feedmeapples) [inline README.md:84]: Let's also update this line to be explicit what it does

```suggestion
$ yarn install
```

File: README.md [hunk 2]
```diff
@@ -1,45 +1,122 @@
-# Website
+# Temporal documentation repository
 
-This website is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.
+This repo contains Temporal's open source documentation.
```
Comment (rylandg) [inline README.md:3]: Why open source?

File: README.md [hunk 3]
```diff
@@ -1,45 +1,122 @@
-# Website
+# Temporal documentation repository
```
Comment (rylandg) [inline README.md:1]: don't put repository, it's obviously a repository :)

File: README.md [hunk 4]
```diff
@@ -1,45 +1,122 @@
...
 
-This website is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.
+This repo contains Temporal's open source documentation.
 
-### Prerequisites
+You can view the published documentation at https://docs.temporal.io.
 
-Install [YARN](https://classic.yarnpkg.com/en/docs/install/).
+This repo utilizes the [Docusaurus 2](https://v2.docusaurus.io/) framework, a modern static website generator, which allows you to build and view the documentation locally as you contribute.
```
Comment (rylandg) [inline README.md:7]: Don't use the word allow. Allow is bad because its what a parent does to a child. Use words that empower like "enable" or "makes it possible" even a neutral "provides". 


```suggestion
This repo utilizes the [Docusaurus 2](https://v2.docusaurus.io/) framework, a modern static website generator, which makes it possible to build and view the documentation locally.
```

Comment (rylandg) [reply README.md:7]: That being said, hot reloading isn't exactly a unique feature to docusaurus

File: README.md [hunk 5]
```diff
@@ -1,45 +1,122 @@
...
 
-### Installation
+## Contribution guidelines
+
+Contributions from the community are highly encouraged.
```
Comment (rylandg) [inline README.md:11]: Make these two sentences on a single line.

File: README.md [hunk 6]
```diff
@@ -1,45 +1,122 @@
...
+
+To contribute please follow these steps.
+
+### Step 1: Fork and clone
+
+To ensure that each change is properly peer reviewed, we request that you follow standard open source contribution processes.
```
Comment (rylandg) [inline README.md:17]: Is there a link we can provide?

File: README.md [hunk 7]
```diff
@@ -1,45 +1,122 @@
...
+
+This means that you should start by forking the project into your own GitHub account.
```
Comment (rylandg) [inline README.md:19]:
```suggestion
Start by forking the project into your own GitHub account and then cloning that fork onto your local machine.
```

Remove other sentence.

File: README.md [hunk 8]
```diff
@@ -1,45 +1,122 @@
...
 
 ```
-$ yarn
+$ git clone git@github.com:<your-account>/documentation-legacy.git
 ```
 
-### Local Development
+### Step 2: Add upstream remote
+
+Then add the original repository as a new remote to your local clone.
```
Comment (rylandg) [inline README.md:29]:
```suggestion
Using git, add the original repository as a new remote to your local clone.
```

File: README.md [hunk 9]
```diff
@@ -1,45 +1,122 @@
...
-$ yarn start
+$ cd documentation-legacy
+$ git remote add upstream git@github.com/temporal/documentation-legacy.git
 ```
 
-This command starts a local development server and open up a browser window. Most changes are reflected live without having to restart the server.
+### Step 3: Create a local branch
 
-### Build
+When you are ready to start making changes, sync your local clone with the upstream project and your forked GitHub project.
```
Comment (rylandg) [inline README.md:38]: This is a heavy sentence.

File: README.md [hunk 10]
```diff
@@ -1,45 +1,122 @@
...
 
 ```
-$ yarn build
+$ git checkout master
+$ git pull upstream master && git push origin master
+```
+
+Then, create a branch to work on.
```
Comment (rylandg) [inline README.md:45]:
```suggestion
Create a branch to work on.
```

Comment (rylandg) [reply README.md:45]: I think you could use "then" less especially since it's already structured into steps.

File: README.md [hunk 11]
```diff
@@ -1,45 +1,122 @@
...
+
+```
+$ git checkout -b your-branch
```
Comment (rylandg) [inline README.md:48]: don't know if this is actually needed right? Why do we care if they work off master in their fork.

Comment (rylandg) [reply README.md:48]: Also can you add languages to the snippets

ie "```bash

     ```"

File: README.md [hunk 12]
```diff
@@ -1,45 +1,122 @@
...
+```
+
+Now you can make your changes. See the next section about building and viewing the site locally for those steps.
```
Comment (rylandg) [inline README.md:51]: This sentence could be a lot more digestible.

File: README.md [hunk 13]
```diff
@@ -1,45 +1,122 @@
...
+
+```
+git push -u origin your-branch
+```
+
+This will create the branch on your GitHub project fork.
+
+**Note** that the -u flag only needs to be used the first time you create a pull request for this branch.
+
+Next, visit the project fork in your GitHub account and locate the new branch, and click "Compare and pull request".
```
Comment (rylandg) [inline README.md:65]: At this point I would either include pictures or defer to a guide of contributing to another repo.

File: README.md [hunk 14]
```diff
@@ -1,45 +1,122 @@
...
+
+Ensure the "base fork" points to the correct repository and branch and provide a clear title and explanation of the changes with links to any issues that are being fixed by the change.
```
Comment (rylandg) [inline README.md:67]: Never used the term "base fork" until here. What is it?

File: README.md [hunk 15]
```diff
@@ -1,45 +1,122 @@
...
+
+Then click "Create pull request".
+
+## Building and viewing the site locally
+
+This website is built using [Docusaurus 2](https://v2.docusaurus.io/).
+
+Check out their [installation requirements](https://v2.docusaurus.io/docs/installation#requirements) for tooling prerequisites.
+
+### Step 1: Initialize yarn
```
Comment (rylandg) [inline README.md:77]: Tbh I don't think we need to explain yarn. If people want to learn about yarn they can use google etc. Just provide a 1 sentence summary and the commands IMO.

Comment (rylandg) [reply README.md:77]: Probably can just comment the commands themselves inline if you want to provide more context.

File: README.md [hunk 16]
```diff
@@ -1,45 +1,122 @@
...
 
-This command generates static content into the `build` directory and can be served using any static contents hosting service.
+### Step 2: Build the website
 
-### Versioning
+The following command generates static content into the `build` directory which can be served using any static contents hosting service.
 
 ```
-$ yarn run docusaurus docs:version 1.1.0
+$ yarn build
```
Comment (rylandg) [inline README.md:92]: You should never need to use this right?

File: README.md [hunk 17]
```diff
@@ -1,45 +1,122 @@
...
 
 ```
-$ GIT_USER=<Your GitHub username> USE_SSH=true yarn deploy
+$ yarn start
 ```
 
-If you are using GitHub pages for hosting, this command is a convenient way to build the website and push to the `gh-pages` branch.
+This command starts a local development server and opens up a browser window.
+
+Most changes to the site are reflected instantly without having to restart the server.
```
Comment (rylandg) [inline README.md:107]: What changes aren't?

File: README.md [hunk 18]
```diff
@@ -1,45 +1,122 @@
...
+
+You can run `yarn start` independently of `yarn build`.
```
Comment (rylandg) [inline README.md:109]: Remove 'build' references entirely IMO

File: README.md [hunk 19]
```diff
@@ -1,45 +1,122 @@
...
+
+## When will my changes be reflected in docs.temporal.io?
+
+As soon as your pull request is merged, a new build will automatically kick off and deploy the changes to the [next](https://docs.temporal.io/docs/next/) version of the docs.
```
Comment (rylandg) [inline README.md:113]: Might as well mention you get a preview on PR submission

File: README.md [hunk 20]
```diff
@@ -1,45 +1,122 @@
...
+
+Temporal will accumulate changes in the next version and periodically release them to the "current" version of the docs.
```
Comment (rylandg) [inline README.md:115]: Can you reword this to indicate that our team manually does this and it's not some automatic CRON (periodically is ambiguous basically)

File: README.md [hunk 21]
```diff
@@ -1,45 +1,122 @@
...
+
+Documentation version schemas will reflect the Temporal API version that they are pinned to as well as the iteration of the documentation for that API version.
```
Comment (rylandg) [inline README.md:117]: Maybe take a screenshot of where this info is displayed on the site. The moment I read schemas as a user I'm going to zone out.

Comment (cullywakelin) [reply README.md:117]: I will once we adopt that schema and are using it live. At present, it would be confusing because we aren't doing that yet.

File: README.md (hunk 14 above)
Comment (cullywakelin) [reply README.md:67]: This is a term used when comparing a branch which has an upstream remote attached -
It only shows up during the comparison and right before you click Create pull request.
This guide has a screen shot. https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/
I will grab a screen shot the next time I create a branch and compare - and add it to this README.

File: README.md (hunk 15 above)
Comment (cullywakelin) [reply README.md:77]: Since you had to explain to me that yarn start is actually executed via a custom script, and because it is not in their CLI docs, then I don't want to assume the reader knows anything about standard yarn usage. I would prefer to keep this information in the README.

File: README.md (hunk 16 above)
Comment (cullywakelin) [reply README.md:92]: removing yarn build

File: README.md (hunk 7 above)
Comment (cullywakelin) [reply README.md:19]: These are two separate steps. Since it is followed by a command, one could be confused and assume that the command does both.

In this case it is better to have two separate sentences to address each step in the process.

I will add a screen shot after the first sentence -

File: README.md (hunk 11 above)
Comment (cullywakelin) [reply README.md:48]: Creating a branch is good practice. It also allows someone to work on multiple things at once. It is a step that is recommended in all of the contribution guides I have read including https://github.com/firstcontributions/first-contributions

File: README.md (hunk 13 above)
Comment (cullywakelin) [reply README.md:65]: Linking to another guide.

File: .forestry/front_matter/templates/standard.yml [hunk 22]
```diff
@@ -52,4 +52,6 @@ fields:
       format: markdown
   label: Release Version
 pages:
+- blog/temporal-transparency-update.md
```
[the remark this answers is not in the document]
Comment (cullywakelin) [reply .forestry/front_matter/templates/standard.yml:55]: I accidentally pulled this change from another commit when I rebased - 
Ya, this is not related to this PR. This change is already on master I think.

File: README.md [hunk 23]
```diff
@@ -1,60 +1,62 @@
...
+### Fork and clone
 
-### Step 1: Fork and clone
+To ensure that each change is relevant and properly peer reviewed, we request that you adhere to open source contribution best practices. GitHub's [first-contributions repo README](https://github.com/firstcontributions/first-contributions) offers a decent overview.
 
-To ensure that each change is properly peer reviewed, we request that you follow standard open source contribution processes.
+Start by forking the project into your own GitHub account.
 
-This means that you should start by forking the project into your own GitHub account.
+![Fork repository button](static/img/readme/forkrepo.png)
```
[the remark this answers is not in the document]
Comment (cullywakelin) [reply README.md:21]: See if this improves. I increased the zoom in the browser and took the screen shot on the mac screen vs my 1080 x 720 screen.

File: README.md [hunk 24]
```diff
@@ -1,60 +1,62 @@
...
 
-Then, clone your own fork of the repo onto your local machine.
+Using git, clone your fork onto your local machine.
```
[the remark this answers is not in the document]
Comment (cullywakelin) [reply README.md:23]: I like your suggestion to reference git. No harm in being clear.

File: README.md [hunk 25]
```diff
@@ -68,55 +70,52 @@ Ensure the "base fork" points to the correct repository and branch and provide a
...
-The following command generates static content into the `build` directory which can be served using any static contents hosting service.
-
-```
-$ yarn build
+```bash
+$ yarn install
 ```
 
-It also ensures that the site is building correctly without errors or misconfigurations.
+Read the [yarn CLI docs](https://classic.yarnpkg.com/en/docs/cli/) to learn what yarn is doing.
```
[the remark this answers is not in the document]
Comment (cullywakelin) [reply README.md:91]: Adopted the "if you want" part but I prefer placing the conjunction after the statement subject.

File: README.md [hunk 26]
```diff
@@ -1,45 +1,125 @@
...
+
+![Netlify build preview](static/img/readme/netlifypreview.png)
+
+As soon as your pull request is merged, a new build will automatically kick off and deploy the changes to the [next](https://docs.temporal.io/docs/next/) version of the docs.
+
+Temporal documentation will accumulate changes in the "next" version and be periodically released by our team to the "current" version of the docs.
+
+Documentation version schemas will reflect the Temporal API version that they are pinned to as well as the iteration of the documentation for that API version.
+
+Version schema: `{temporal api version}-docs-{docs version}`
```
Comment (rylandg) [inline README.md:121]: Can you make these two colon items bold

**key**: value

File: README.md [hunk 27]
```diff
@@ -1,45 +1,125 @@
...
+
+This will fail if there are source code errors or references to relative paths that do not exist.
+
+Most changes to the site are reflected instantly without having to restart the server. Installing a new plugin will require you to restart.
+
+## When will my changes be reflected in docs.temporal.io?
+
+You will have a chance to preview the changes in your pull request. Click "Details" next to the Netlify deploy-preview check.
+
+![Netlify build preview](static/img/readme/netlifypreview.png)
```
Comment (rylandg) [inline README.md:113]: It seems nit picky but the image has a link in the picture which actually confused me enough that I tried to click it. Can you just take this pic without the link at the end.
